src/routes/guilds/guild/leaderboard.ts

```diff
@@ -0,0 +1,35 @@
+import { Route, RouteStore } from 'klasa-dashboard-hooks';
+import ApiRequest from '../../../lib/structures/api/ApiRequest';
+import ApiResponse from '../../../lib/structures/api/ApiResponse';
+import { ratelimit, iteratorRange, fetchAllLeaderboardEntries } from '../../../lib/util/util';
+
+export default class extends Route {
+
+	public constructor(store: RouteStore, file: string[], directory: string) {
+		super(store, file, directory, { name: 'guildLeaderboard', route: 'guilds/:guild/leaderboard' });
+	}
+
+	@ratelimit(2, 2500)
+	public async get(request: ApiRequest, response: ApiResponse) {
+		const guildID = request.params.guild;
+
+		const guild = this.client.guilds.get(guildID);
+		if (!guild) return response.error(400);
+
+		const limit = 'limit' in request.query ? Number(request.query.limit) : 10;
+		if (!Number.isInteger(limit) || limit <= 0 || limit > 100) return response.error(400);
+
+		const before = 'before' in request.query ? Number(request.query.before) : null;
```

Comment:
```suggestion
		const before = Number(request.query?.before) || null
```

Same as above although I'm not sure if this also works because the `||` goes outside of the `Number()`? Haven't really done a lot of optional chaining yet.